sycl/CMakeLists.txt

```diff
@@ -14,6 +14,10 @@ if(MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
 endif()
 
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wno-error=terminate")
+endif()
+
```

Comment:
I don't think it's a good idea to enable this unconditionally.
I suggest making a CMake option controlling this behavior and make it OFF by default.
Similar to LLVM_ENABLE_WERROR.

To keep our sources clean from warnings, you can enable this option in buildbot scripts.

Reply:
Done

Reply:
Can't we just add "-DLLVM_ENABLE_WERROR=ON" in builtbot config?

Reply:
> Can't we just add "-DLLVM_ENABLE_WERROR=ON" in builtbot config?

If it works for external project, it's probably preferred.
NOTE: Ivan added a couple of other options in addition to -Werror.